builder/vsphere/common/step_add_cdrom.go

```diff
@@ -73,27 +73,29 @@ func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multiste
 		}
 	}
 
-	ui.Say("Mounting ISO images...")
 	if path, ok := state.GetOk("iso_remote_path"); ok {
-		if err := vm.AddCdrom(s.Config.CdromType, path.(string)); err != nil {
-			state.Put("error", fmt.Errorf("error mounting an image '%v': %v", path, err))
-			return multistep.ActionHalt
-		}
+		s.Config.ISOPaths = append(s.Config.ISOPaths, path.(string))
 	}
 
 	// Add our custom CD, if it exists
 	if cd_path, _ := state.Get("cd_path").(string); cd_path != "" {
 		s.Config.ISOPaths = append(s.Config.ISOPaths, cd_path)
 	}
 
-	if len(s.Config.ISOPaths) > 0 {
-		for _, path := range s.Config.ISOPaths {
-			if err := vm.AddCdrom(s.Config.CdromType, path); err != nil {
-				state.Put("error", fmt.Errorf("error mounting an image '%v': %v", path, err))
-				return multistep.ActionHalt
-			}
+	ui.Say("Mounting ISO images...")
+	// Bug in govmomi: can't batch-create cdroms and then mount ISO files, that
+	// resuls in wrong UnitNumber. So do that one-by-one.
```

Comment:
If calling out a bug (or is it actually a limitation) in `vmware/govmomi`, it would be ideal to include a reference so that this can be reviewed in the future.

Reply:
Well… I currently don't have resources for creating a minimal testcase for this, so let's just rename this to a "limitation" if that's okay with you

Reply:
Renamed per comment above